### PR TITLE
Fixing some strong mode warnings and errors

### PR DESCRIPTION
**Pull request by srawlins**
All tests pass.

### CODE REVIEW

**Comment by ochafik** [comment]
Hi Sam, could you please address the warning that made travis to fail?

```
[warning] The argument type '(MirroredEntity<Declaration>) → dynamic' cannot be assigned to the parameter type '(MirroredEntity<RuleSet>) → void' (/home/travis/build/google/dart-scissors/lib/src/css_mirroring/rulesets_processor.dart, line 38, col 32)
```

(you can run `TEST_EXAMPLES=0 ./scripts/presubmit.sh` to replicate Travis' tests)

File: lib/eager_transformer.dart [hunk 1]
```diff
@@ -36,7 +36,7 @@ class _ScissorsSettings extends SettingsBase
 }
 
 List<List<Transformer>> _createPhases(_ScissorsSettings settings) {
-  var phases = [
+  var phases = <List>[
```
Comment (ochafik) [inline lib/eager_transformer.dart]: `<List<Transformer>>`

File: lib/src/css_mirroring/mirrored_entities.dart [hunk 2]
```diff
@@ -58,7 +58,7 @@ class MirroredEntity<T extends TreeNode> {
     return node is MediaDirective || node is HostDirective;
   }
 
-  MirroredEntities<dynamic> getChildren(List<dynamic> getEntityChildren(T _)) {
+  MirroredEntities<T> getChildren(List<T> getEntityChildren(T _)) {
```
Comment (ochafik) [inline lib/src/css_mirroring/mirrored_entities.dart]: Children are typically of a different type, please introduce a type param:

```
  MirroredEntities<C> getChildren/*<C>*/(List/*<C>*/ getEntityChildren(T _)) {
```

File: lib/src/css_mirroring/rulesets_processor.dart [hunk 3]
```diff
@@ -29,7 +29,7 @@ RemovalResult editFlippedRuleSet(MirroredEntity<RuleSet> mirroredRuleSet,
     Direction flippedDirection, BufferedTransaction trans) {
   final subTransaction = trans.createSubTransaction();
 
-  MirroredEntities<Declaration> mirroredDeclarations = mirroredRuleSet
+  MirroredEntities<RuleSet> mirroredDeclarations = mirroredRuleSet
```
Comment (ochafik) [inline lib/src/css_mirroring/rulesets_processor.dart]: This change seems wrong, declarations is a `List<Declaration>` (see comment above re/ making getChildren generic)

File: lib/src/utils/process_utils.dart [hunk 4]
```diff
@@ -42,7 +42,7 @@ Future<ProcessResult> pipeInAndOutOfNewProcess(Process p, input) async {
   if (input is String) {
     p.stdin.write(input);
   } else if (input is List) {
-    p.stdin.add(input);
+    p.stdin.add(input as List<int>);
```
Comment (ochafik) [inline lib/src/utils/process_utils.dart]: Could you please just test `input is List<int>` above instead?

Comment (srawlins) [inline lib/src/utils/process_utils.dart]: done

File: lib/src/css_mirroring/rulesets_processor.dart (hunk 3 above)
Comment (srawlins) [inline lib/src/css_mirroring/rulesets_processor.dart]: done

File: lib/src/css_mirroring/mirrored_entities.dart (hunk 2 above)
Comment (srawlins) [inline lib/src/css_mirroring/mirrored_entities.dart]: done

File: lib/eager_transformer.dart (hunk 1 above)
Comment (srawlins) [inline lib/eager_transformer.dart]: done

**Comment by srawlins** [comment]
OK, presubmit.sh says `SUCCESS: good to go!`

**Comment by ochafik** [comment]
Thanks a lot!

**Pull request merged**